pkg/volume/persistent_claim/persistent_claim.go

```diff
@@ -80,6 +81,19 @@ func (plugin *persistentClaimPlugin) NewBuilder(spec *volume.Spec, pod *api.Pod,
 		return nil, err
 	}
 
+	// If a GID annotation is provided set the GID attribute.
+	if volumeGid, ok := pv.Annotations["GID"]; ok {
```

Comment:
The annotation should be a constant value and prefixed with the Kube namespace:  `kubernetes.io/supplemental-gid` or something similar.

Docs will need to be updated, too, describing this functionality and are relevant to this PR.